askama_derive/src/generator.rs

```diff
@@ -1,21 +1,18 @@
 use crate::heritage::{Context, Heritage};
 use crate::input::{Print, Source, TemplateInput};
 use crate::parser::{parse, Cond, CondTest, Expr, Loop, Node, Target, When, Whitespace, Ws};
-use crate::{
-    filters, get_template_source, read_config_file, CompileError, Config, WhitespaceHandling,
-};
+use crate::{get_template_source, read_config_file, CompileError, Config, WhitespaceHandling};
 
-use proc_macro2::TokenStream;
+use proc_macro::TokenStream;
 use quote::{quote, ToTokens};
 
 use std::collections::HashMap;
 use std::path::{Path, PathBuf};
 use std::{cmp, hash, mem, str};
 
 /// The actual implementation for askama_derive::Template
-#[doc(hidden)]
-pub fn derive_template(input: TokenStream) -> TokenStream {
-    let ast: syn::DeriveInput = syn::parse2(input).unwrap();
+pub(crate) fn derive_template(input: TokenStream) -> TokenStream {
```

Comment:
Can be a simple `pub` since proc-macro crates can't export anything other than proc macros anyways.

Reply:
I'd prefer to use `pub(crate)` anyway, to make it clear what's happening to the human reader.

Reply:
I think `#![deny(unreachable_pub)]` had a problem with `pub`, but I can be mistaken. IMO `pub(crate)` is clearer.